nni/nas/oneshot/pytorch/strategy.py

```diff
@@ -144,3 +147,92 @@ class RandomOneShot(OneShotStrategy):
 
     def __init__(self, **kwargs):
         super().__init__(RandomSamplingLightningModule, **kwargs)
+
+    def attach_model(self, base_model):
+        # This may conflict with the `self.run`` method. In the `run`, the base_model and evaluator are wrapped in `Model`. 
+        from nni.nas.oneshot.pytorch.base_lightning import traverse_and_mutate_submodules
+        _reason = 'The reason might be that you have used the wrong execution engine. Try to set engine to `oneshot` and try again.'
+
+        if not isinstance(base_model, nn.Module):
+            raise TypeError('Model is not a nn.Module. ' + _reason)
+        
+        extra_mixed_ops = getattr(base_model, "get_extra_mixed_oprations")() if hasattr(base_model, "get_extra_mixed_oprations") else []
```

Comment:
I think you don't need `get_extra_mixed_operations`. This brings unnecessary coupling between strategy and model space. And you spelled `operations` wrong.

Instead, you can get the mutation classes from the mutation hooks (see https://stackoverflow.com/questions/961048/get-class-that-defined-method ). And then, retrieve a list of mixed operations.

We might also change the interface of mutation hooks a little bit in future, to make things more elegant.